tests/testthat/test-map_n.R

```diff
@@ -60,3 +60,17 @@ test_that("pmap on data frames performs rowwise operations", {
   expect_is(pmap_chr(mtcars2, paste), "character")
   expect_is(pmap_raw(mtcars2, function(mpg, cyl) as.raw(cyl)), "raw")
 })
+
+test_that("preserves S3 class", {
+  fctr <- factor("a")
```

Comment:
I think you only need to test a single S3 class here, since the other options don't cause the tests to exercise any more code.